src/modules/nationalId/index.ts

```diff
@@ -1,43 +1,50 @@
+/**
+ * Check allCharactersSame
+ * @category all Characters Same
+ * @method allCharactersSame
+ * @param  {String?}          inputString [String of numbers - like this: 1111111111]
+ * @return {Boolean}                    [valid or no]
+ */
+
+function allCharactersSame(inputString?: string): boolean {
+	if (!inputString){
+		return false;
+	}
+
+	const n = inputString?.toString()?.length;
+	for (let i = 1; i < n; i++)
+		if (inputString[i] !== inputString[0])
+			return false;
+
+	return true;
+}
+
+
 /**
  * Check National-id validation
  * @category National id
  * @method verifyIranianNationalId
- * @param  {String?}          nationalId [String of national id - like this: 0018465986]
+ * @param  {String?}          nationalCode [String of national id - like this: 0018465986]
  * @return {Boolean}                    [valid or no]
  */
 
-function verifyIranianNationalId(nationalId?: string | number): boolean | undefined {
-	if (!nationalId) return;
-	let code = nationalId.toString();
-	const codeLength = code.length;
-	const notAllowedDigits = {
-		"0000000000": true,
-		"1111111111": true,
-		"2222222222": true,
-		"3333333333": true,
-		"4444444444": true,
-		"5555555555": true,
-		"6666666666": true,
-		"7777777777": true,
-		"8888888888": true,
-		"9999999999": true,
-	};
-
-	if (code in notAllowedDigits) return false;
-	if (codeLength < 8 || codeLength > 10) return false;
-	code = ("00" + code).substring(codeLength + 2 - 10);
-	if (+code.substring(3, 9) === 0) return false;
-
-	const lastNumber = +code.substring(9);
-	let sum = 0;
-
-	for (let i = 0; i < 9; i++) {
-		sum += +code.substring(i, i + 1) * (10 - i);
+function verifyIranianNationalId(nationalCode?: string | number): boolean | undefined {
+	if (!nationalCode){
+		return false
 	}
 
-	sum = sum % 11;
+	let temp = nationalCode.toString();
 
-	return (sum < 2 && lastNumber === sum) || (sum >= 2 && lastNumber === 11 - sum);
+	if (!nationalCode || nationalCode?.toString().length !== 10 || allCharactersSame(temp)) {
+		return false;
+	}
+	temp = `0000${temp}`.substr(temp?.length + 4 - 10);
+	if (parseInt(temp.substr(3, 6), 10) === 0) return false;
+	const c = parseInt(temp.substr(9, 1), 10);
+	let s = 0;
+	for (let i = 0; i < 9; i++) s += parseInt(temp.substr(i, 1), 10) * (10 - i);
+	s %= 11;
+	return (s < 2 && c === s) || (s >= 2 && c === 11 - s)
```

Comment:
I propose:
- renaming `c` and `s` to some meaningful variables
- extracting this part into a function with a meaningful name. This is hard to understand and debug.